main.tf

```diff
@@ -141,27 +141,33 @@ module "eks" {
   }
 
   ################################################################################
-  # Handle BYO IAM policy
+  # Handle BYO IAM Roles & Policies
   ################################################################################
-  create_iam_role                                = var.cluster_iam_role_name == null ? true : false   # v17: manage_cluster_iam_resources
-  iam_role_name                                  = var.cluster_iam_role_name                          # v17: cluster_iam_role_name
+  # BYO - EKS Cluster IAM Role
+  create_iam_role = var.cluster_iam_role_arn == null ? true : false
```

Comment:
Curious what create_iam_role is being used for? I don't see references to it at this level elsewhere. Looks like a different instance of it appears in eks_managed_node_group_defaults below,

Reply:
The 'create_iam_role' determines whether a an IAM role is created for the EKS cluster or to use an existing IAM role. Likewise for the workers IAM role.